### ISSUE TITLE
SqlLab page can't handle UTF-8 codec

### ISSUE THREAD

**Issue by kevinwen2k** [apache/superset#16574]
Hi all, I would like to report another bug here: The SqlLab can’t handle the UTF-8 codec. Steps to reproduce:

1. Run a query that can retrieve data from foreign country and I got the following error as the first 2 screenshots (either sync query or async query).
2. When I used other software tool (including browser) to run the same query, we can get the result displayed as the 3rd and 4th screenshots

Without the problem solved, we can’t create charts with data including character in UTF-8 or even other format. It would be great to get it fixed in current 1.2 branch before we can upgrade to 1.3. Thanks.

**Comment by kevinwen2k** [comment]
![Screen Shot 2021-09-02 at 3 41 28 PM](https://user-images.githubusercontent.com/4207782/131926245-b2f3f105-390a-4162-987a-e9630375bbc9.png)
![Screen Shot 2021-09-02 at 3 42 09 PM](https://user-images.githubusercontent.com/4207782/131926251-0c8a0721-ac3a-4ba2-9b8d-7f1e977574d3.png)
![Screen Shot 2021-09-02 at 3 39 43 PM](https://user-images.githubusercontent.com/4207782/131926258-442ab326-8954-4952-b9b2-29c161493196.png)
![Screen Shot 2021-09-02 at 3 39 55 PM](https://user-images.githubusercontent.com/4207782/131926263-4aecf2c1-f3ea-4f4f-810c-138d95fe5622.png)

**Comment by etr2460** [comment]
According to the error messages, this seems to be an issue with Snowflake, and not with sql lab. Was this a regression that used to work in superset, or did you only now notice the issue. I would be inclined to think that there's not an issue with superset here, given that i've used utf-8 characters with Presto fine, but let me know if you think differently before I close the issue

**Comment by spdustin** [comment]
`0xe9` is a bytestring; perhaps it should be cast `to_char(column, 'utf-8')` first? Or, more appropriately, correct the data stored in Snowflake…because it seems like some entries are storing data in that column as a binary rather than a UTF-8-encoded string.

**Comment by rumbin** [comment]
Could you please try to reproduce this issue with a minimal example which does not depend on any existing table? Something like:. 
`select 'éèëē ñäâæ öøœ' as unicode_chars`
  
This way we can narrow down, if @spdustin's suspicion  is true and anyone with a Snowflake DB can reproduce the bug, if it still exists.

**Comment by kevinwen2k** [comment]
@rumbin Yes, when running the query you provided, I could get the result without any problem. However, the input from all of the editors in any software is compliant with UTF-8 and can't reproduce the scenario that happened on our end where data came from different sources or are collected from 3rd party. It's hard to enforce all of our sources to convert the non-UTF8 bytes to UTF8 bytes. 

Also, as I mentioned in this ticket earlier, Snowflake (for some reasons) can convert these byte string back to UTF-8 characters and displayed well in the browser. DBVisualizer does similar thing that can convert byte into some non-human readable characters, but at least it won't cause the error when running the query. 

The following query goes through in Chrome browser connecting Snowflake account (meaning that Snowflake implemented both frontend and backend for the endpoint service with our Snowflake account) and in DBvisualizer, but it throws an error in SqlLab. 

```
select install_country, install_city 
from my_table
where return_date between current_date - 10 and current_date - 1 
    and install_country = 'Venezuela' and endswith(install_city, 'rida') 
limit 100;  
```

I know the data are not perfect, but what I wanted was to have the query goes through, so we can create chart on dashboard. I'm ok if those non-human readable characters  are displayed in the charts.

![Screen Shot 2021-09-09 at 9 42 57 AM](https://user-images.githubusercontent.com/4207782/132729022-d7c5875a-d107-4590-bf3d-be79bd3f0f5b.png)
![Screen Shot 2021-09-09 at 9 43 27 AM](https://user-images.githubusercontent.com/4207782/132729048-8286676a-b0ef-4fe7-a534-d78efe4148ab.png)
![Screen Shot 2021-09-09 at 9 41 16 AM](https://user-images.githubusercontent.com/4207782/132729065-0b0fe6ea-c323-4edf-a1dd-8c9759343958.png)

**Comment by rusackas** [comment]
Is anyone still facing this in current-era Superset? Tempted to close this due to inactivity, but I don't want to close it if it's still causing issues.

**Comment by rusackas** [comment]
Closing this as stale since it's been silent for so long, and we're trying to steer toward a more actionable Issues backlog. If people are still encountering this in current versions (currently 3.x) please re-open this issue, open a new Issue with updated context, or raise a PR to address the problem. Thanks!
